src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -141,6 +144,14 @@ private Task StartAsyncCore()
             return _startTcs.Task;
         }
 
+        private async Task<NegotiationResponse> GetNewNegotiationResponse()
```

Comment:
~~New~~

It doesn't know it's "new" ;). The method just gets a negotiation response